rapid_response_xblock/static/js/src/rapid.js

```diff
@@ -117,13 +172,32 @@
 
           var tspanText = tspan.text();
           tspan.text(tspanText + " " + word);
-          if (tspan.node().getComputedTextLength() > barWidth) {
+          if (tspan.node().getComputedTextLength() > hypotenuse) {
             // The new word would go beyond the bar width boundary,
             // so change tspan back to its old text and create one with the
             // new word on a new line.
+
+            // If there is only one long word we don't have a choice but to render it
+            if (tspanText.length === 0) {
+              return;
+            }
+
+            if (currentLine === 1) {
+              // there is a maximum of two lines until we add the ellipses
+              tspan.text(tspanText + "...");
+              currentLine++;
+              return;
+            }
+
+            if (currentLine > 1) {
+              // maximum of two lines
+              tspan.text(tspanText);
+              return;
+            }
```

Comment:
If `currentLine > 1`, it should be an immediate return a la `!word` at line 113.

Reply:
Line # reference was kinda silly. Basically I'm suggesting that we should get rid of this `currentLine > 1` clause and immediately return, so the beginning of the loop would look something like this:

```javascript
        words.forEach(function(word) {
          // If the word is blank (i.e.: input text is empty), or if we have exceeded our maximum number of lines
          // don't do anything.
          if (!word || currentLine > 1) {
            return;
          }
          //...
        })
```